velox/exec/HashBuild.cpp

```diff
@@ -85,9 +85,6 @@ HashBuild::HashBuild(
   }
 
   // Identify the non-key build side columns and make a decoder for each.
-  const auto numDependents = outputType->size() - numKeys;
```

Comment:
In the test case you mentioned, what is the number of `numDependents`? I guess the reserve function fails only when the reserved value is non-positive, am I correct?

Reply:
@rui-mo，
If the value < 0, will trigger exception.

```
terminate called after throwing an instance of 'std::length_error'
  what():  vector::reserve
Aborted

```

Reply:
I guess a better solution would be to check whether the reserved value is greater than 0. If yes, the reserve functions can be called.

Reply:
If-Else branch is not better for CPU performance.
And here, reserve will not bring benefit because  output schema field limited.

Reply:
To reserve before use the vector is a common code style in Velox, and they have gave me some comments on that before. I think we'd better follow that. As for the performance, I think `if else` would not affect the performance too much here.

Reply:
Advantages to use reserve: https://www.geeksforgeeks.org/using-stdvectorreserve-whenever-possible/.